match/lib/match/runner.rb

```diff
@@ -329,6 +359,23 @@ def fetch_provisioning_profile(params: nil, certificate_id: nil, app_identifier:
     end
     # rubocop:enable Metrics/PerceivedComplexity
 
+    def find_matching_profile(profiles, certificates, spaceship = nil)
+      profiles.filter do |profile_path|
+        # Check if it matched any of the certificates
+        found = false
+        certificates.each do |certificate|
+          UI.verbose("Checking if profile '#{profile_path}' includes certificate '#{certificate['FileName']}'")
+          next unless FastlaneCore::ProvisioningProfile.includes_certificate?(profile_path: profile_path, certificate: certificate)
+          UI.verbose("Checking if profile is valid on dev portal")
+          next unless 
```

Comment:
this seems like it should do something to check if the profile is valid, or did i miss something?